RUFAS/routines/animal/ration/animal_requirements.py

```
@@ -263,25 +263,95 @@ def calc_rqmts(BW, MW, DOP, animal_type, parity=None, CI=None , TP_Milk=None, Fa
     # The sum of dry matter intake of each feed is assumed to be less than
     # dry matter intake estimation (Sum of Feed < DMIest).
     # --------------------------------------------
-    if animal_type == 'cow':
-        # [A.Cow.D.1]
-        # Fat corrected milk (kg)
-        FCM = (0.4 * Milk) + (15 * Fat_Milk * (Milk / 100))
-        # [A.Cow.D.2]
-        # Dry matter intake estimation (kg)
-        if lactating:
-            DMIest = (0.372 * FCM + 0.0968 * BW ** 0.75) * (1 - math.exp(-0.192 * ((DIM / 7) + 3.67)))
-        else:
-            DMIest = ((1.97 - 0.75 * math.exp(0.16 * (DOP - 280))) / 100) * BW
-    elif animal_type == 'heifer':
-        #TODO: Actual calculation for DMIest
-        DMIest = 0
+    if AnimalBase.config['NASEM_or_NRC_requirement_calculations'] == 'NASEM':
+        DMIest = calculate_DMI_NASEM(BW, MW, DIM, lactating, NEl, BCS5, parity)
+    elif AnimalBase.config['NASEM_or_NRC_requirement_calculations'] == 'NRC':
+        DMIest = calculate_DMI_NRC(animal_type, BW, DOP, DIM, lactating, Milk, Fat_Milk)
+    else:
+        DMIest=[]
+        print(f"DMI estimation method \
```

I imagine this should raise an exception and log an error in the OutputManager, but I'm holding off until that is fully implemented.